examples/debug/src/Tools.hs

```diff
@@ -0,0 +1,39 @@
+module Tools where
+
+import Cardano.Api qualified as CAPI
+import Data.Aeson ((.=))
+import Data.Aeson qualified as JSON
+import Data.Text (Text)
+import GHC.Natural (Natural)
+import Ledger (Address (Address), PubKeyHash)
+import Ledger.Tx.CardanoAPI (toCardanoAddress)
+import Plutus.V1.Ledger.Api (Credential (PubKeyCredential))
+import Prelude
+
+pkhFromHash :: String -> PubKeyHash
+pkhFromHash key =
+  let res = JSON.fromJSON $ JSON.object ["getPubKeyHash" .= key]
```

Comment:
Instead of using `fromJSON` directly and then pattern-matching on the , you can use `Data.Aeson.Types.parseEither` (maybe it's also exported from `Data.Aeson`):

```haskell
pkhFromHash = either error id . JSON.parseEither (parseJSON @PubKeyHash) $ ...
```

This way the error message might be more informative

It might also be helpful to add a comment explaining why constructing a JSON `Value` like this is necessary

Reply:
Yeah why are we doing this? PubKeyHash has a fromString instance, is that not sufficient?

Reply:
fixed